lib/httplib/httplib.go

```diff
@@ -41,15 +41,27 @@ type HandlerFunc func(w http.ResponseWriter, r *http.Request, p httprouter.Param
 // StdHandlerFunc specifies HTTP handler function that returns error
 type StdHandlerFunc func(w http.ResponseWriter, r *http.Request) (interface{}, error)
 
+// ErrorWriter is a function responsible for writing the error into response
+// body.
+type ErrorWriter func(w http.ResponseWriter, err error)
+
 // MakeHandler returns a new httprouter.Handle func from a handler func
-func MakeHandler(fn HandlerFunc) httprouter.Handle {
+func MakeHandler(fn HandlerFunc, errWriter ...ErrorWriter) httprouter.Handle {
+	if len(errWriter) > 1 {
+		panic("at most 1 ErrorWriter can be passed to MakeHandler")
```

Comment:
`panic` seems a bit extreme here given that we support passing a list of `ErrorWriter` - can we just take the first one instead? Is there something I missed?

Reply:
Yes, I agree. If one wanted to solve this in Go, one option is using two APIs instead of abusing the slice arguments:
```go
func MakeHandler(fn HandlerFunc) { ... }
func MakeHandlerWithErrorWriter(fn HandlerFunc, errWriter ErrorWriter) { ... }
```
and just have `MakeHandler` invoke the other with `trace.WriteError`.

Reply:
Good idea, better to enforce this at compile time rather than runtime.